File: sdks/php/src/Models/PurchaseOrder.php
<?php

namespace Puxbay\Models;

class PurchaseOrder
{
    public function __construct(
        public readonly string $id,
        public readonly string $poNumber,
        public readonly string $supplier, // ID
        public readonly ?string $supplierName = null,
        public readonly string $status,
        public readonly float $totalAmount,
        public readonly ?\DateTimeImmutable $expectedDeliveryDate = null,
        public readonly array $items,
        public readonly \DateTimeImmutable $createdAt,
        public readonly \DateTimeImmutable $updatedAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            poNumber: $data['po_number'],
            supplier: $data['supplier'],
            supplierName: $data['supplier_name'] ?? null,
            status: $data['status'],
            totalAmount: (float) $data['total_amount'],
            expectedDeliveryDate: isset($data['expected_delivery_date']) ? new \DateTimeImmutable($data['expected_delivery_date']) : null,
            items: $data['items'] ?? [], // Could use POItem DTOs here
            createdAt: new \DateTimeImmutable($data['created_at']),
            updatedAt: new \DateTimeImmutable($data['updated_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'po_number' => $this->poNumber,
            'supplier' => $this->supplier,
            'status' => $this->status,
            'total_amount' => $this->totalAmount,
        ];
    }
}

File: sdks/dotnet/src/Models/PurchaseOrder.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class PurchaseOrder
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("po_number")]
        public string PoNumber { get; set; }

        [JsonProperty("supplier")]
        public string Supplier { get; set; }

        [JsonProperty("supplier_name")]
        public string SupplierName { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("total_amount")]
        public double TotalAmount { get; set; }

        [JsonProperty("expected_delivery_date")]
        public DateTime? ExpectedDeliveryDate { get; set; }

        // Keeping items as dynamic dictionary for simplicity in POCO if strict item type not defined
        [JsonProperty("items")]
        public List<Dictionary<string, object>> Items { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime? UpdatedAt { get; set; }
    }
}
